src/DIRAC/Core/Security/m2crypto/X509Chain.py

```diff
@@ -830,8 +838,7 @@ def generateChainFromRequestString(self, pemData, lifetime=86400, requireLimited
         # You can't request a limit proxy if you are yourself not limited ?!
         # I think it should be a "or" instead of "and"
         limited = requireLimited and self.isLimitedProxy().get("Value", False)
-
-        return self.generateProxyToString(lifetime, diracGroup, 1024, limited, req.get_pubkey())
+        return self.generateProxyToString(lifetime, diracGroup, None, limited, req.get_pubkey())
```

Comment:
Are you sure of this?

```
In [2]: M2Crypto.RSA.gen_key?
Signature:
M2Crypto.RSA.gen_key(
    bits,
    e,
    callback=<function keygen_callback at 0x7f1d48d665e0>,
)
Docstring:
Generate an RSA key pair.

:param bits: Key length, in bits.

:param e: The RSA public exponent.

:param callback: A Python callable object that is invoked
                 during key generation; its usual purpose is to
                 provide visual feedback. The default callback is
                 keygen_callback.

:return: M2Crypto.RSA.RSA object.
File:      ~/mambaforge/envs/dirac-development/lib/python3.9/site-packages/M2Crypto/RSA.py
Type:      function
```

Reply:
I am not sure why you are showing me this signature ? 
The key length is not used in that case because we actually give a key as 5ft parameter, so 1024 was never used there. I replaced it with `None` to avoid confusion

Reply:
Ahh right!